### PR TITLE
Added YIELDONE Bid Adapter for Prebid.js 1.0

### PR DESCRIPTION
**Pull request by takuhou**
## Type of change
- [x] New bidder adapter 

## Description of change
New bidder adapter added for YIELDONE

- test parameters for validating bids
```
{
  bidder: 'yieldone',
  params: { 
    placementId: '44082'
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](//webdemo.dac.co.jp/pone/hb_test_multi.html?pbjs_debug=true) sample page.

- contact email of the adapter’s maintainer y1dev@platform-one.co.jp
- [x] official adapter submission

### CODE REVIEW

**Comment by takuhou** [comment]
Hello jsnellbaker

I'm sorry. Since it was setting retargeting, it was not delivered.
Please check again.

**Comment by jsnellbaker** [comment]
Hello @takuhou Thanks for making the change.  I checked the test placement with the `44082` and it's working consistently now.  However the test placement `44084` (the one that's provided in the .md file) still isn't working for me.  Was that supposed to be fixed as well?  

Alternatively if you want to use `44082` for overall testing, can you update the md file?

**Comment by takuhou** [comment]
Hello jsnellbaker

I changed placementId to 44082.
Please check again.

Thanks.

**Comment by takuhou** [comment]
Hello jsnellbaker

CI seems to have failed, but do you know why?
Do I need to fix something?

Thanks.

**Comment by jsnellbaker** [comment]
@takuhou Thanks for the change.  I restarted the Travis build and it seems to have passed this time.  Everything looks good now from my end.

File: modules/yieldoneBidAdapter.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Test Parameters
+```
+  var adUnits = [{
+    code: 'banner-ad-div',
+    sizes: [300, 250],
+    bids: [{
+    bidder: 'yieldone',
+      params: { 
+        placementId: '44082',
+        width:'300',
```
Comment (mkendall07) [inline modules/yieldoneBidAdapter.md]: why do you need repeat width/height params when you can get this off the bid.sizes?

**Comment by mkendall07** [comment]
@takuhou 
everything looks good except for the width/height. Can you comment on that?

**Comment by takuhou** [comment]
Hello @mkendall07 

This width and height are basically used as parameters when requesting to the server, specifying the size of the inventory.

There are special cases and you may distribute multiple size inventory to 300x250 inventory.
For example, 300 x 225 ads may be delivered at 300 x 250 inventory. At that time, have you specify width: 300 / height: 225.For that reason, we have width and height parameters as YIELDONE.

**Comment by mkendall07** [comment]
so in your example, 300x225 is an extra size that can deliver in addition to the 300x250 right? but it doesn't look like you read the adUnit sizes at all, so you only pass 300x225.

**Comment by takuhou** [comment]
Hello @mkendall07 

Well. Explanation is difficult.
OK. In HB in Prebid.js this case may not exist, so change it to take size from bid.

Please wait as it fixes.

Thanks.

**Comment by takuhou** [comment]
Hello @mkendall07

I fixed it, so please check it.

File: modules/yieldoneBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['y1'],
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.placementId);
+  },
+  buildRequests: function(validBidRequests) {
+    return validBidRequests.map(bidRequest => {
+      const params = bidRequest.params;
+      const width = bidRequest.sizes[0];
```
Comment (mkendall07) [inline modules/yieldoneBidAdapter.js]: this won't work with multi size arrays. Example sizes = [[300,250], [728,90]]

take a look at https://github.com/prebid/Prebid.js/blob/master/src/utils.js#L113 for help with parsing.

Comment (takuhou) [reply modules/yieldoneBidAdapter.js]: Hello @mkendall07 

Please tell me what kind of case is multi-size with same Placement ID.

Comment (mkendall07) [reply modules/yieldoneBidAdapter.js]: you don't have to read multiple sizes if you dont want, but you should take the first size if it's multiple.

Comment (takuhou) [reply modules/yieldoneBidAdapter.js]: Hello @mkendall07

I fixed it, so please check it.

**Comment by takuhou** [comment]
Hello @mkendall07 

Correspondence of bids.sizes is completed, please confirm.

Thanks.

**Comment by mkendall07** [comment]
This is merged into master. Please submit a PR to the [docs repo](https://github.com/prebid/prebid.github.io) to add a file for your adapter to the [bidders directory](https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders) so your adapter's params will appear on the [bidders page](http://prebid.org/dev-docs/bidders.html). Thank you for contributing

**Pull request merged**